src/components/EventDetailsPage.tsx

```diff
@@ -0,0 +1,49 @@
+import { ReactElement, useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { fetchEvent } from "../client/client";
+
+export type EventDetailsPageProps = {
+  event?: Event;
+};
+
+const EventDetailsPage = ({ event }: EventDetailsPageProps): ReactElement => {
+  const { id } = useParams();
+  const [loading, setLoading] = useState(false);
```

Comment:
I'm wondering instead of having a `loading` and `setLoading` useState we could have a useState that tracks the current event of this page, and if the data is loading then the event is null, if the data has loaded then that event contains that data

Reply:
Yess this was right!